packages/financial-templates-lib/src/price-feed/TraderMadePriceFeed.js

```diff
@@ -0,0 +1,278 @@
+
+    this.priceFeedDecimals = priceFeedDecimals;
+    this.ohlcPeriod = ohlcPeriod;
+
+    this.convertPriceFeedDecimals = number => {
+      // Converts price result to wei
+      // returns price conversion to correct decimals as a big number
+      return this.toBN(parseFixed(number.toString(), priceFeedDecimals).toString());
+    };
+  }
+
+  getCurrentPrice() {
+    return this.currentPrice;
+  }
+
+  getHistoricalPrice(time, verbose = false) {
+    if (this.lastUpdateTime === undefined) {
+      return null;
+    }
+
+    // Set first price time in `historicalPrices` to first non-null price.
+    let firstPriceTime;
+    for (let p in this.historicalPricesMinute) {
+      if (this.historicalPricesMinute[p] && this.historicalPricesMinute[p].openTime) {
+        firstPriceTime = this.historicalPricesMinute[p];
+        break;
+      }
+    }
+
+    // If there are no valid price time, return null.
+    if (!firstPriceTime) {
+      return null;
+    }
+
+    // If the time is before the first piece of data in the set, return null because
+    // the price is before the lookback window.
+    if (time < firstPriceTime.openTime) {
+      return null;
+    }
+
+    // historicalPrices are ordered from oldest to newest.
+    // This finds the first priceTime whose closeTime is after the provided time.
+    const match = this.historicalPricesMinute.find(price => {
+      return time < price.closeTime;
+    });
+
+    // If there is no match, that means that the time was past the last data point.
+    // In this case, the best match for this price is the current price.
+    if (match === undefined) {
+      let returnPrice = this.currentPrice;
+      if (verbose) {
+        console.group(`\n(${this.pair}) No OHLC available @ ${time}`);
+        console.log(
+          `- ✅ Time is later than earliest historical time, fetching current price: ${this.web3.utils.fromWei(
+            returnPrice.toString()
+          )}`
+        );
+        console.log(
+          `- ⚠️  If you want to manually verify the specific exchange prices, you can make a GET request to: \n- https://marketdata.tradermade.com/api/v1/live?currency=${this.pair}&api_key={api-key}`
+        );
+        console.groupEnd();
+      }
+      return this.currentPrice;
+    }
+
+    let returnPrice = match.closePrice;
+    if (verbose) {
+      console.group(`\n(${this.pair}) Historical OHLC @ ${match.closeTime}`);
+      console.log(`- ✅ Close Price:${this.web3.utils.fromWei(returnPrice.toString())}`);
+      console.log(
+        `- ⚠️  If you want to manually verify the specific exchange prices, you can make a GET request to: \n- https://marketdata.tradermade.com/api/v1/timeseries?currency=${this.pair}&api_key={api-key}&start_date=${time}&end_date=${match.closeTime}&format=records&interval=minute&period=${this.ohlcPeriod}`
+      );
+      console.groupEnd();
+    }
+    return returnPrice;
+  }
+
+  getHistoricalPricePeriords() {
+    return this.historicalPricesHourly;
```

Comment:
@daywiss does this work for dev mining? This returnsr the hourly periods for `getHistoricalPricePeriods` and uses the minute periods for `getHistoricalPrice`

Reply:
yes that should work